frontend/tests/builders/partyBuilder.js

```diff
@@ -30,13 +30,20 @@ export class PartyBuilder {
   }
 
   build() {
-    if (this.players === []) {
+    if (this.players && this.players.length === 0) {
       this.players = [
-        new PlayerBuilder(`a ${this.party} player`).build(),
-        new PlayerBuilder(`another ${this.party} player`).build()
+        new PlayerBuilder(`a ${this.party} player`)
+          .withParty(this.party)
+          .build(),
+        new PlayerBuilder(`another ${this.party} player`)
+          .withParty(this.party)
+          .build()
       ];
+    } else {
+      this.players.forEach(player => {
+        player.party = this.party;
+      });
```

Comment:
Good catch! I suggest we run the _party assignment_ code in any case to simplify a bit:

```suggestion
      this.players = [
        new PlayerBuilder(`a ${this.party} player`).build(),
        new PlayerBuilder(`another ${this.party} player`).build()
      ];
    }
    this.players.forEach(player => {
      player.party = this.party;
    });
```

Reply:
Yep, looks decent 😄